### PR TITLE
 Array: Inverse of an array

### LINKED ISSUE #3043: ARRAY : Inverse of an array
## please assign me this issue

This is very standard problem, in which we swaps the indices of the array with element.
This type of basic questions can make the users understand of the feel and concept of array.

### Have you read the [Contributing Guidelines on Pull Requests]
YES

I will be implementing the problem code in CPP with proper comments and explaination along with test cases.
code will be mine , I can guarantee that.

### PR DESCRIPTION
**Pull request by Op-panda**
This is inverse of an array problem ,quiet famous and important one. please do merge this problem

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes 

### Description
Inverse of an array problem, 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#3043

### CODE REVIEW

File: C-Plus-Plus/ds/inverse of an array.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+
+
+    Sample input
+    3
+    0 1 2
+
+    Sample Output
+    0 1 2
+    */
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/inverse of an array.cpp]: Not properly indented. Check the existing code to understand how to do that

File: C-Plus-Plus/ds/inverse of an array.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            inv[j] = i;         //after storing values of array in j
+                                //at inverse array we store corresponding indices of array arr[].
+    }
+
+    for(int i =0 ; i<n; i++){
+        cout<<inv[i]<<" ";      //output the inverse array
+    }
+
+    return 0;
+    }
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/inverse of an array.cpp]: Format the Code properly and indent it

File: C-Plus-Plus/ds/inverse of an array.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        cin>>arr[i];            //putting values in array
+    }
+     for(int i =0 ; i<n; i++){
+            int j = arr[i];     //j is temporary variable
+            inv[j] = i;         //after storing values of array in j
+                                //at inverse array we store corresponding indices of array arr[].
+    }
+
+    for(int i =0 ; i<n; i++){
+        cout<<inv[i]<<" ";      //output the inverse array
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/inverse of an array.cpp]: Never use comments beside the code. Use it above the line of code

File: C-Plus-Plus/ds/inverse of an array.cpp [hunk 4]
```diff
@@ -0,0 +1,50 @@
+
+    //INVERSE OF AN ARRAY
+
+    /*If the array elements are swapped with their corresponding indices,
+      the array finally results is inverse of the array.
+    */
+
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/inverse of an array.cpp]: Add a more intuitive description

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Op-panda** [comment]
@HarshCasper please help me out updating readme. Can I create another pull request for that because its not accepting now from the same branch as many files have added in the origianal repo. And I have made the code changes please review that

**Comment by Op-panda** [comment]
@HarshCasper Now please Review this I have fixed the changes you requested. please do a review once more and tell if I need to do some more changes??

File: C-Plus-Plus/README.md [hunk 5]
```diff
@@ -122,6 +122,7 @@
 - [Ascending Priority Queue] (ds/Ascending_priority_queue.cpp)
 - [Priority_Queue_using_Linked_List](ds/Priority_Queue_using_Linked_List.cpp)
 - [Maximum Width of Binary Tree](ds/Maximum_Width_BinaryTree.cpp)
+- [Inverse of an array](C-Plus-Plus/ds/inverse of an array.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: This will not work. Fix it!

File: C-Plus-Plus/ds/inverse of an array.cpp [hunk 6]
```diff
@@ -0,0 +1,73 @@
+/*
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/inverse of an array.cpp]: Name your file as `inverse_of_array.cpp`

File: C-Plus-Plus/README.md (hunk 5 above)
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: This will not work. Fix it!

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Indent and format the code as well properly

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Indent and format the code as well properly

File: C-Plus-Plus/ds/inverse of an array.cpp (hunk 6 above)
Comment (Op-panda) [reply C-Plus-Plus/ds/inverse of an array.cpp]: do i need to make a different PR with another file name. please help out. I dont know how to do that

File: C-Plus-Plus/ds/inverse_of_an_array.cpp [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+/*
+If the array elements are swapped with their corresponding indices,
+the array finally results is inverse of an array.
+
+For Example,
+If we have 4,0,2,3,1 in original array with indices 0,1,2,3,4 respectively,
+so assume a blank array of same size. Now at 4th index, element in inverse array will be 0. 
+Similarly, at 0th index, element in inverse array will be 1
+and similar swapping for the rest of elemnts 
+At last you will be getting 1,4,2,3,0 in inverse array.
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/inverse_of_an_array.cpp]: Remove this

File: C-Plus-Plus/ds/inverse_of_an_array.cpp [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+QUESTION-->
+
+Input:
+1)First line contains t, number of test cases.
+2)Second line contains size of the original array.
+3)Third line contains elements of the original array.
+
+Output:
+1)contains t outputs, inverse array of each original array. */
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/inverse_of_an_array.cpp]: Remove this

File: C-Plus-Plus/ds/inverse_of_an_array.cpp [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+
+#include<bits/stdc++.h>
+using namespace std;
+int main(){
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/inverse_of_an_array.cpp]: Keep the main code logic outside the `main()` function. Use `main()` only for the driver code

File: C-Plus-Plus/ds/inverse_of_an_array.cpp [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+    int j = ori[i];         
+    inv[j] = i;         
+    }
+for(int i=0; i<n; i++)
+//output the inverse array
+    {
+    cout<<inv[i]<<" ";      
+    }
+}
+return 0;
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/inverse_of_an_array.cpp]: None of the code here is either formatted or indented. How come you resolved my comments?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/ds/inverse_of_an_array.cpp (hunk 10 above)
Comment (Op-panda) [reply C-Plus-Plus/ds/inverse_of_an_array.cpp]: I am sorry. I didn't know about those. This is my first contibution.

**Comment by Op-panda** [comment]
Is now the changes ok ??

File: C-Plus-Plus/ds/inverse_of_an_array.cpp (hunk 9 above)
Comment (HarshCasper) [reply C-Plus-Plus/ds/inverse_of_an_array.cpp]: Not yet resolved @Op-panda

File: C-Plus-Plus/ds/inverse_of_an_array.cpp [hunk 11]
```diff
@@ -0,0 +1,65 @@
+/*
+INVERSE OF AN ARRAY   
+/*
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/inverse_of_an_array.cpp]: Remove the `/*`

File: C-Plus-Plus/ds/inverse_of_an_array.cpp [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+
+#include <bits/stdc++.h>
+using namespace std;
+
+// ALGORITHM -->
+// create two arrays, ori and inv.
+// give input for ori array.
+// create a temporary variable then, and store the element from the original array.
+// In inv array for each value of temp, store the index of ori array.
+// Display the inverse array.
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/inverse_of_an_array.cpp]: Remove this

File: C-Plus-Plus/ds/inverse_of_an_array.cpp [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+        int inv[n];
+
+        for (int i = 0; i < n; i++)
+            cin >> ori[i];
+
+        inverse(ori, n, inv);
+    }
+    return 0;
+}
+// Driver code ends
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/inverse_of_an_array.cpp]: Remove this

File: C-Plus-Plus/ds/inverse_of_an_array.cpp [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+5
+4 0 2 3 1
+3
+0 1 2
+
+Sample Output
+1 4 2 3 0
+0 1 2
+*/
+
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/inverse_of_an_array.cpp:52]: Remove this blank line

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**